sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/uninstall/ResourceCleanupStep.java

```diff
@@ -0,0 +1,50 @@
+package com.mesosphere.sdk.scheduler.uninstall;
+
+import com.mesosphere.sdk.offer.OfferRecommendation;
+import com.mesosphere.sdk.offer.ResourceUtils;
+import com.mesosphere.sdk.offer.UninstallRecommendation;
+import com.mesosphere.sdk.scheduler.plan.PodInstanceRequirement;
+import com.mesosphere.sdk.scheduler.plan.Status;
+
+import java.util.Collection;
+import java.util.Optional;
+
+/**
+ * Step which implements the uninstalling of a particular reserved resource. For instance, persistent
+ * volumes and cpu.
+ */
+public class ResourceCleanupStep extends UninstallStep {
+    /**
+     * Creates a new instance with the provided {@code resourceId} and initial {@code status}.
+     */
+    public ResourceCleanupStep(String resourceId, Status status) {
```

Comment:
I think it was done this way to not be passing the `StateStore` down to the individual resource cleaning steps (this is just a rename of `UninstallStep` so I could make a baseclass).